Engine/source/core/resource.h

```diff
@@ -194,7 +194,7 @@ template<class T> class ResourceHolder : public ResourceHolderBase
 {
 public:
    ResourceHolder(T *t) : ResourceHolderBase(t) {}
-   virtual ~ResourceHolder() { delete ((T*)mRes); }
+   virtual ~ResourceHolder() { mRes = static_cast<T*>(mRes); SAFE_DELETE(mRes); }
```

Comment:
I would strongly suggest adding a unit test to make sure ~ResourceHolder actually calls the destructor correctly considering this is undefined behavior and could easily break without knowing.